### DISCUSSION TITLE
no package installation possible

### DISCUSSION THREAD

**Discussion by Feedback1000** [IronLanguages/ironpython3#1752] [Q&A]
Hey.

I tried to install sympy.
 
Previously I installed IronPython 3.4.
 
The `ipy -m pip install sympy==1.4` command didn't work because I received the following error message:

> Der Befehl "ipy" ist entweder falsch geschrieben oder
konnte nicht gefunden werden..

![Bild_2023-11-11_125120183](https://github.com/IronLanguages/ironpython3/assets/148341502/d290948c-9d06-477c-804e-ba434f837d30)

Even though the path exists:
![Screenshot 2023-11-11 125350_2](https://github.com/IronLanguages/ironpython3/assets/148341502/40fc3b01-e79e-4eb9-9c0e-4003e7a06751)

Can you help?

Regards,
Fabian

**Comment by BCSharp** [comment]
`ipy.exe` on your system is installed in directory `C:\Programme\IronPython 3.4` but you try to execute it from directory `C:\Windows\System32`, therefore the executable is not found. Change the current dirtectory to the IronPython isnstallation directory and run from there. E.g.:
```cmd
C:\Windows\System32>cd "\Programme\IronPython 3.4"
C:\Programme\IronPython 3.4>ipy -m pip install sympy==1.4
```
To be able to run `ipy` from anywhere, add the IronPython installation directory to the `PATH` environment variable. Adding the subdirtectory `Scripts` may be useful too because it's where `isympy.exe` is going to be installed.

**Comment by Feedback1000** [comment]
Hey Pavel.

 

Thank you for your help. But it doesn’t matter if I want to install sympy or f.e. vectors I get the same error:

 

 



 

Can you help?

 

Regards

Fabian

 

Von: Pavel Koneski ***@***.***> 
Gesendet: Sonntag, 12. November 2023 20:24
An: IronLanguages/ironpython3 ***@***.***>
Cc: Feedback1000 ***@***.***>; Author ***@***.***>
Betreff: Re: [IronLanguages/ironpython3] no package installation possible (Discussion #1752)

 

ipy.exe on your system is installed in directory C:\Programme\IronPython 3.4 but you try to execute it from directory C:\Windows\System32, therefore the executable is not found. Change the current dirtectory to the IronPython isnstallation directory and run from there. E.g.:

C:\Windows\System32>cd "\Programme\IronPython 3.4"
C:\Programme\IronPython 3.4>ipy -m pip install sympy==1.4

To be able to run ipy from anywhere, add the IronPython installation directory to the PATH environment variable. Adding the subdirtectory Scripts may be useful too because it's there where isympy.exe is going to be installed.

**Comment by slozier** [comment]
If ipy is not on your path, depending on your shell, you may have to do `.\ipy -m pip install sympy==1.4`.

**Comment by Feedback1000** [comment]
Not yet helpful:

 



 

 

Von: slozier ***@***.***> 
Gesendet: Mittwoch, 13. Dezember 2023 16:49
An: IronLanguages/ironpython3 ***@***.***>
Cc: Feedback1000 ***@***.***>; Author ***@***.***>
Betreff: Re: [IronLanguages/ironpython3] no package installation possible (Discussion #1752)

 

If ipy is not on your path, depending on your shell, you may have to do .\ipy -m pip install sympy==1.4.

**Comment by BCSharp** [comment]
@Feedback1000 maybe you do not realize, but your recent posts arrive without the error messages you are getting. Without the error messages it is hard to guess what is going wrong on your system.

One thought comes to mind: since the default installation folder used by the MSI installer (`C:\Programme\IronPython 3.4` in your case) is a read-only location for a regular user, to update it with new packages you have to run `ipy -m pip install sympy==1.4` from the elevated (administrator) command prompt, rather than the regular shell prompt. Have you tried that?
